File: 2305-fair-distribution-of-cookies/2305-fair-distribution-of-cookies.cpp
class Solution {
public:
    int ans ;
    void go(vector<int>&c,vector<int>&temp,int idx){
        if(idx==c.size()){
            int res =0;
            for(int i=0;i<temp.size();++i){
                res=max(res,temp[i]);
            }
            ans = min(ans,res);
            return ;
        }
        for (int i=0;i<temp.size();++i){
            temp[i]+=c[idx];
            go(c,temp,idx+1);
            temp[i]-=c[idx];
        }
        
    }
    int distributeCookies(vector<int>& cookies, int k) {
        ans =INT_MAX;
        vector<int>temp(k);
        go(cookies,temp,0);
        return ans;
    }
};

File: 2305-fair-distribution-of-cookies/2305-fair-distribution-of-cookies.py
class Solution:
    def distributeCookies(self, c: List[int], k: int) -> int:
        ans = 1e18
        def go(temp,idx):
            nonlocal ans
            if(idx==len(c)):
                res =0
                for x in temp:
                    res=max(res,x)
                ans = min(ans,res)
                return
            for i in range(len(temp)):
                temp[i]+=c[idx]
                go(temp,idx+1)
                temp[i]-=c[idx]
                if(temp[i]==0):
                    break
        temp = [0 for x in range(k)]
        go(temp,0)
        return ans
    